src/security/Handshake.cc

```diff
@@ -550,23 +551,26 @@ Security::HandshakeParser::ParseCertificate(const SBuf &raw, Security::CertPoint
     pCert = nullptr;
     debugs(83, 2, "TLS parsing is not supported without OpenSSL. " << raw);
 #endif
+    return pCert;
```

Comment:
uhm, that is the point of smart pointers. not to leak the memory when the local code or some called code throws like that.